src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/AutoCrafterListener.java

```diff
@@ -58,6 +65,21 @@ public void onInteract(PlayerRightClickEvent e) {
                     return;
                 }
 
+                // Check if the recipe of the item is disabled.
+                boolean unlocked = true;
+                for (Recipe recipe : Bukkit.getRecipesFor(e.getItem())) {
+                    if (!e.getPlayer().hasDiscoveredRecipe(((Keyed) recipe).getKey())) {
+                        unlocked = false;
+                        break;
+                    }
+                }
+
+                if (e.getPlayer().getWorld().getGameRuleValue(GameRule.DO_LIMITED_CRAFTING) && !unlocked) {
+                    e.cancel();
```

Comment:
Calling `e.cancel()` here is redundant. The event is cancelled anyway after this, so you can just cancel it *before* the if-statement and remove the other two calls. It is called unconditionally here.

Reply:
I will just remove the redundant `e.cancel()`.